internal/test/framework/framework.go

```diff
@@ -0,0 +1,198 @@
+	WinrmClient *winrm.Client
+	// remoteDir is the directory to which files will be transferred to, on the Windows VM
+	RemoteDir string
+	// SSHClient contains the ssh client information to access the Windows VM via ssh
+	SSHClient *ssh.Client
+	// CloudProvider holds the information related to cloud provider
+	cloudProvider cloudprovider.Cloud
+	// k8sclientset is the kubernetes clientset we will use to query the cluster's status
+	K8sclientset *kubernetes.Clientset
+}
+
+// Setup sets up the initial test cluster for the Windows
+// TODO: move this to return error and do assertions there
+func (f *TestFramework) Setup() {
+	if err := f.createWindowsVM(); err != nil {
+		log.Fatalf("failed to create Windows VM: %v", err)
+	}
+	// TODO: Add some options to skip certain parts of the test
+	if err := f.setupWinRMClient(); err != nil {
+		log.Fatalf("failed to setup winRM client for the Windows VM: %v", err)
+	}
+	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
+	// in the list of services
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := f.configureOpenSSHServer(); err != nil {
+		log.Fatalf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	}
+	if err := f.createRemoteDir(); err != nil {
+		log.Fatalf("failed to create remote dir with error: %v", err)
+	}
+	if err := f.getSSHClient(); err != nil {
+		log.Fatalf("failed to get ssh client for the Windows VM created: %v", err)
+	}
+	if err := f.getKubeClient(); err != nil {
+		log.Fatalf("failed to get kube client with error: %v", err)
+	}
+}
+
+// getKubeClient setups the kubeclient that can be used across all the test suites.
+func (f *TestFramework) getKubeClient() error {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return fmt.Errorf("could not build config from flags: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("could not create k8s clientset: %v", err)
+	}
+	f.K8sclientset = clientset
+	return nil
+}
+
+// createWindowsVM spins up the Windows VM in the given cloud provider and gives us the credentials to access the
+// windows VM created
+func (f *TestFramework) createWindowsVM() error {
+	// TODO: Add validation for the environment variables to see if they're properly populated or not
+	kubeconfig := os.Getenv("KUBECONFIG")
+	awsCredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	artifactDir := os.Getenv("ARTIFACT_DIR")
+	privateKeyPath := os.Getenv("KUBE_SSH_KEY_PATH")
+	// Use Windows 2019 server image with containers in us-east1 zone for CI testing.
+	// TODO: Move to environment variable that can be fetched from the cloud provider
+	imageID := "ami-0b8d82dea356226d3"
+	instanceType := "m4.large"
```

Comment:
Please see #106 and use `m5a.large`

Reply:
Thanks for catching it, I thought I already did this while doing rebase but obviously not and this could be causing the failures we're noticing in CI